nominal/core/workbook.py

```diff
@@ -1,21 +1,49 @@
 from __future__ import annotations
 
+import logging
 from dataclasses import dataclass, field
-from typing import Protocol
+from enum import Enum
+from typing import Mapping, Protocol, Sequence
 
 from nominal_api import scout, scout_notebook_api
-from typing_extensions import Self
+from typing_extensions import Self, deprecated
 
+from nominal._utils.dataclass_tools import update_dataclass
 from nominal.core._clientsbunch import HasScoutParams
 from nominal.core._utils import HasRid
 
+logger = logging.getLogger(__name__)
+
+
+class WorkbookType(Enum):
+    WORKBOOK = "WORKBOOK"
+    COMPARISON_WORKBOOK = "COMPARISON_WORKBOOK"
+
+    @classmethod
+    def _from_conjure(cls, workbook_type: scout_notebook_api.NotebookType) -> WorkbookType:
+        if workbook_type.value == "WORKBOOK":
+            return cls.WORKBOOK
+        elif workbook_type.value == "COMPARISON_WORKBOOK":
+            return cls.COMPARISON_WORKBOOK
+        else:
+            raise ValueError(f"Unknown workbook type: {workbook_type}")
+
+    def _to_conjure(self) -> scout_notebook_api.NotebookType:
+        return {
+            "WORKBOOK": scout_notebook_api.NotebookType.WORKBOOK,
+            "COMPARISON_WORKBOOK": scout_notebook_api.NotebookType.COMPARISON_WORKBOOK,
+        }[self.value]
+
 
 @dataclass(frozen=True)
 class Workbook(HasRid):
     rid: str
     title: str
     description: str
-    run_rid: str | None
+    run_rids: Sequence[str] | None
+    asset_rids: Sequence[str] | None
```

Comment:
Let's document that these are generally mutually exclusive, which is why they're optional